File: medium/2026-06/516. Longest Palindromic Subsequence.go
package main

import "fmt"

func main() {
	longestPalindromeSubseq("total")
}

func longestPalindromeSubseq(s string) int {
	n := len(s)
	prev := make([]int, n)
	curr := make([]int, n)
	for i := 0; i < 0; i++ {
		prev = append(prev, 0)
		curr = append(curr, 0)
	}
	fmt.Println(prev, curr)
	return 0
}

File: medium/2026-06/516. Longest Palindromic Subsequence.py
import pprint
class Solution:
    def longestPalindromeSubseq(self, s: str) -> int:
        n = len(s)
        prev = [0] * n
        curr = [0] * n

        for i in range(n-1, -1, -1):
            curr[i] = 1
            for j in range(i+1, n):
                if s[i] == s[j]:
                    curr[j] = prev[j-1] + 2
                else:
                    curr[j] = max(prev[j], curr[j-1])

            prev = curr
            curr = [0] * n
            print(prev)
s = Solution()
res = s.longestPalindromeSubseq("total")


